maps_dashboards/public/model/layerRenderController.ts

```diff
@@ -79,13 +95,48 @@ export const prepareDataLayerSource = (
 };
 
 export const handleDataLayerRender = (
-  mapLayer: MapLayerSpecification,
+  mapLayer: DocumentLayerSpecification,
   mapState: MapState,
   services: MapServices,
   maplibreRef: MaplibreRef,
   beforeLayerId: string | undefined
 ) => {
-  return prepareDataLayerSource(mapLayer, mapState, services).then((result) => {
+  const filters: Filter[] = [];
+  const geoField = mapLayer.source.geoFieldName;
+  const geoFieldType = mapLayer.source.geoFieldType;
+
+  // geo bounding box query supports geo_point fields
+  if (
+    geoFieldType === 'geo_point' &&
+    mapLayer.source.useGeoBoundingBoxFilter &&
+    maplibreRef.current
+  ) {
+    const mapBounds = maplibreRef.current.getBounds();
+    const filterBoundingBox = {
+      bottom_right: {
+        lon: adjustLongitudeForSearch(mapBounds.getSouthEast().lng),
```

Comment:
Will calling mapBounds.getSouthEast().wrap() solves this problem?

Reply:
@VijayanB that won't work out of the box, calling `wrap()` will wrap the longitude to range `(-180, 180]`. In the screenshot, after wrapping, the longitude of left bounding is great than the right bounding, the consequence is OpenSearch won't return results from the expected bounding box. What do people usually do to tackle this issue in a map application?

<img width="1597" alt="image" src="https://user-images.githubusercontent.com/486382/210466499-6e6c73dc-f66d-462a-b7f4-3a931ae995cf.png">
<img width="500" alt="image" src="https://user-images.githubusercontent.com/486382/210466517-014e390b-4dd3-4358-85bd-bba3a286dd7d.png">

Reply:
For this did we break the query? I hope no because geobounding box query takes care of these things in the code.

Reply:
@navneet1v i raised new PR to support corner cases here: https://github.com/opensearch-project/dashboards-maps/pull/175/ . Here, i didn't break it into multiple query, instead wrapped the bounds. However, i had put ceiling to MIN/MAX if more than at least one complete map is visible. ( like hybrid approach ) Let me know whether you see any problem with that approach.